packages/ast/src/lowering.candy

```diff
@@ -958,14 +958,16 @@ fun lowerFunctionType(
       })
 
   let returnType = cst.child.returnType
-      .map<Maybe<AstType>>({ lowerType(context, id, it) })
-      .orElse({
-        context.addError(ExpectedReturnTypeInFunctionTypeCompilerError(context.file, cst))
-        None<AstType>()
-      })
+      .flatMap<AstType>({ lowerType(context, id, cst.child.returnType.unwrap()) })
+  if returnType is None {
+    context.addError(ExpectedReturnTypeInFunctionTypeCompilerError(context.file, cst))
+    return None<AstFunctionType>()
+  }
```

Comment:
This error also gets emitted if the CST contains a return type, but this type couldn't be lowered.